src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -4,8 +4,7 @@ namespace NCronJob;
 
 internal sealed class JobRegistry
 {
-    private readonly HashSet<JobDefinition> allJobs
-        = new(JobDefinitionEqualityComparer.Instance);
+    private readonly List<JobDefinition> allJobs = [];
```

Comment:
Isn't the HashSet easier to understand?

Reply:
It doesn't work. As soon as we mutate the `JobDefinition` the hashcode change and it cannot find it anymore in its internal buckets.

Took me some time to figure that out.

Basically, the following pseudo code doesn't work with a HashSet

```
HashShet.Add(jobDef)
RuntimeRegistry.DisableJob(jobDef)
Assert.True(HashSet.Contains(jobDef)
```